test/functional/cmdLineTests/SystemPropertiesTest/src/SysPropTest.java

```diff
@@ -24,49 +24,66 @@
 import java.io.*;
 import java.nio.charset.Charset;
 
-public class SysPropTest 
+public class SysPropTest
 {
-	/* TestVa&#187;lue&#161; */	
-	static final byte[] B = { (byte)0x54, (byte)0x65, (byte)0x73, (byte)0x74,  (byte)0x56, (byte)0x61,(byte) 187, (byte)0x6C, (byte)0x75, (byte)0x65, (byte) 161};	
+	/* TestVa&#187;lue&#161; */
+	static final byte[] B = { (byte)'T', (byte)'e', (byte)'s', (byte)'t', (byte)'V', (byte)'a', (byte)187, (byte)'l', (byte)'u', (byte)'e', (byte)161};
+	static final byte[] Butf8 = { (byte)'T', (byte)'e', (byte)'s', (byte)'t', (byte)'V', (byte)'a', (byte)0xC2, (byte)187, (byte)'l', (byte)'u', (byte)'e', (byte)0xC2, (byte)161};
 
 	public static void main(String args[])
 	{
 		boolean isWindows = false;
 		if ( args.length == 0) {
-			System.out.println("test failed"); 
+			System.out.println("test failed");
 			return;
 		}
 		String argEncoding = args[0];
-		/* check -Dtestkey=Test?Va?lue?  */
+		/* check -Dtestkey=TestVa?lue?  */
 		try {
 			String osEncoding = "";
 			String strTestProp;
 			if (System.getProperty("os.name").contains("Windows")) {
 				isWindows = true;
 			}
+			String verString = System.getProperty("java.version");
+			for (int i = 0; i < verString.length(); i++) {
+				if (!Character.isDigit(verString.charAt(i))) {
+					verString = verString.substring(0, i);
+					break;
+				}
+			}
+			int javaVersion = Integer.decode(verString);
+
+			/* On jdk18+ with JEP 400 UTF-8 by default, the non-ascii characters in the testkey property
+			 * are converted to UTF8 by the test (not the JVM).
+			 */
+			final byte[] expectedB = javaVersion >= 18 && !isWindows ? Butf8 : B;
```

Comment:
```suggestion
			final byte[] expectedB = (javaVersion >= 18) && (!isWindows ? Butf8 : B);
```

Reply:
The suggested bracketing is wrong, but I get your point.